Test: Refactor assigned test classes to use PrameterizedTest

Nakul-Shivaraj opened the pull request:
# Homework 2 Pull Request

## Checklist before creating

- [x] There are no Checkstyle issues.
- [x] There are no SpotBugs issues.
- [x] There are no javadoc errors.
- [x] All tests in the file pass.

_Please explain if there are any issues/errors._

## Questions

### Will you get full credit on this assignment?
Answer: Yes, Claimed classes/methods, Converted methods to ParameterizedTest as assigned and I did not introduce any problems. 

### How long did you spend on the assignment?
Answer: I had less number of tests that needed to be converted to ParameterizedTest so I just took 1-2hrs.

### What problems did you encounter, and how did you deal with them?
Answer: I did not face issue with converting the tests, but faced issue while pushing them to the main, I did take help from TA, peers and Prof.Ellen to get it resolved.

### What help did you give or receive on this assignment (including answering these question)? What was your AI usage?
Answer: I got help from TA and Prof. Ellen on the git related queries. I was able to help peers with their tests. 
No usage of AI in this assignment at any point.

Nakul-Shivaraj committed:
Test: Refactor assigned test classes to use PrameterizedTest

Nakul-Shivaraj committed:
Test: Refactor assigned test classes to use PrameterizedTest

alexanderbodnar committed:
Fix refine check for getSuggestedFileName (#13974)

* Fix refine check for getSuggestedFileName

* Add expandBrackets test, update CHANGELOG

* CHANGELOG update

* Add notNull assert for expandBrackets

* Add issue to CHANGELOG.md

* remove redundant assertNull

* Update CHANGELOG.md

---------

Co-authored-by: Subhramit Basu <subhramit.bb@live.in>

subhramit committed:
Fix minor typo

koppor committed:
Keep label removal (#13980)

espertusnu commented:
What SpotBugs issues did you have?

espertusnu commented on jablib/src/test/java/org/jabref/logic/layout/format/AuthorAndsReplacerTest.java:
```diff
@@ -11,25 +12,16 @@ class AuthorAndsReplacerTest {
+    @ParameterizedTest
+    @CsvSource({
+            "'', ''",
+            "'Someone, Van Something', 'Someone, Van Something'",
```
Good use of single quotes.

espertusnu commented on jablib/src/test/java/org/jabref/logic/layout/format/AuthorAndsReplacerTest.java:
```diff
@@ -11,25 +12,16 @@ class AuthorAndsReplacerTest {
+    @CsvSource({
+            "'', ''",
+            "'Someone, Van Something', 'Someone, Van Something'",
+            "'John Smith and Black Brown, Peter', 'John Smith & Black Brown, Peter'",
```
Please transfer the comments, like you did in the other file.

Nakul-Shivaraj replied:
Done, File updated

espertusnu reviewed (changes requested):
If your teammates haven't comleted their parts, you might take on some of their load.

Siedlerchr committed:
Fix duplicate key error on insertion (#13981)

* Fix duplicate key error on insertion

* add test

* fix empty lines

* just throws

* just throws

paudelritij committed:
Add options to choose groups during import (#13977)

* Add options to choose groups during import

* Remove unused import

OmkarNagare16 committed:
Make Web Search 'Configure API key' searchable in preferences #13929 (#13982)

* Make Web Search 'Configure API key' searchable in preferences

* Fix Checkstyle empty line issues in PreferencesTab and WebSearchTab

* Trigger CI after updating PR description

* Fix failing LocalizationConsistencyTest: add missing 'Custom API key' entry

* Add changelog entry for searchable API key preferences

* Fix list formatting in changelog

* Move changelog entry to Unreleased section

* Trigger CI

* Update changelog wording based on reviewer suggestion

koppor committed:
Update gradle to 9.3.0-jabref-1 (#13719)

dependabot[bot] committed:
Chore(deps): Bump jablib/src/main/resources/csl-styles (#13986)

Bumps [jablib/src/main/resources/csl-styles](https://github.com/citation-style-language/styles) from `0c01a1f` to `02faf1a`.
- [Release notes](https://github.com/citation-style-language/styles/releases)
- [Commits](https://github.com/citation-style-language/styles/compare/0c01a1f6e29e7225399cd36bf185ba738e9247bb...02faf1a0cc0885892363f0a7e4b7188a5ec0c25b)

---
updated-dependencies:
- dependency-name: jablib/src/main/resources/csl-styles
  dependency-version: 02faf1a0cc0885892363f0a7e4b7188a5ec0c25b
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Chore(deps): Bump org.glassfish.jaxb:jaxb-runtime in /versions (#13987)

Bumps org.glassfish.jaxb:jaxb-runtime from 4.0.4 to 4.0.6.

---
updated-dependencies:
- dependency-name: org.glassfish.jaxb:jaxb-runtime
  dependency-version: 4.0.6
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Chore(deps): Bump jablib/src/main/resources/csl-locales (#13989)

Bumps [jablib/src/main/resources/csl-locales](https://github.com/citation-style-language/locales) from `3b74988` to `c72e5c7`.
- [Release notes](https://github.com/citation-style-language/locales/releases)
- [Commits](https://github.com/citation-style-language/locales/compare/3b749889cf9a04e0afefa2a36bc05dd59a4572d1...c72e5c75c7c432c0268543dc76e5d4ac09b6d9d5)

---
updated-dependencies:
- dependency-name: jablib/src/main/resources/csl-locales
  dependency-version: c72e5c75c7c432c0268543dc76e5d4ac09b6d9d5
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Chore(deps): Bump org.bouncycastle:bcprov-jdk18on in /versions (#13988)

Bumps [org.bouncycastle:bcprov-jdk18on](https://github.com/bcgit/bc-java) from 1.81 to 1.82.
- [Changelog](https://github.com/bcgit/bc-java/blob/main/docs/releasenotes.html)
- [Commits](https://github.com/bcgit/bc-java/commits)

---
updated-dependencies:
- dependency-name: org.bouncycastle:bcprov-jdk18on
  dependency-version: '1.82'
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Chore(deps): Bump org.xmlunit:xmlunit-matchers in /versions (#13994)

Bumps [org.xmlunit:xmlunit-matchers](https://github.com/xmlunit/xmlunit) from 2.10.3 to 2.10.4.
- [Release notes](https://github.com/xmlunit/xmlunit/releases)
- [Changelog](https://github.com/xmlunit/xmlunit/blob/main/RELEASE_NOTES.md)
- [Commits](https://github.com/xmlunit/xmlunit/compare/v2.10.3...v2.10.4)

---
updated-dependencies:
- dependency-name: org.xmlunit:xmlunit-matchers
  dependency-version: 2.10.4
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Chore(deps): Bump org.junit.platform:junit-platform-launcher (#13996)

Bumps [org.junit.platform:junit-platform-launcher](https://github.com/junit-team/junit-framework) from 1.13.1 to 1.13.4.
- [Release notes](https://github.com/junit-team/junit-framework/releases)
- [Commits](https://github.com/junit-team/junit-framework/commits)

---
updated-dependencies:
- dependency-name: org.junit.platform:junit-platform-launcher
  dependency-version: 1.13.4
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Chore(deps): Bump jbangdev/jbang-action from 0.131.0 to 0.132.0 (#13990)

Bumps [jbangdev/jbang-action](https://github.com/jbangdev/jbang-action) from 0.131.0 to 0.132.0.
- [Release notes](https://github.com/jbangdev/jbang-action/releases)
- [Commits](https://github.com/jbangdev/jbang-action/compare/v0.131.0...v0.132.0)

---
updated-dependencies:
- dependency-name: jbangdev/jbang-action
  dependency-version: 0.132.0
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Chore(deps): Bump org.apache.commons:commons-text in /versions (#13997)

Bumps [org.apache.commons:commons-text](https://github.com/apache/commons-text) from 1.13.1 to 1.14.0.
- [Changelog](https://github.com/apache/commons-text/blob/master/RELEASE-NOTES.txt)
- [Commits](https://github.com/apache/commons-text/compare/rel/commons-text-1.13.1...rel/commons-text-1.14.0)

---
updated-dependencies:
- dependency-name: org.apache.commons:commons-text
  dependency-version: 1.14.0
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

koppor committed:
Set Dependabot to run weekly on Sundays

dependabot[bot] committed:
Chore(deps): Bump org.openrewrite.rewrite from 7.16.0 to 7.17.0 (#13993)

* Chore(deps): Bump org.openrewrite.rewrite from 7.16.0 to 7.17.0

Bumps org.openrewrite.rewrite from 7.16.0 to 7.17.0.

---
updated-dependencies:
- dependency-name: org.openrewrite.rewrite
  dependency-version: 7.17.0
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

* Include some fixes

* Disable org.openrewrite.staticanalysis.UnnecessaryParentheses

---------

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Oliver Kopp <kopp.dev@gmail.com>

espertusnu commented:
Whern you're ready, please re-request a review. https://teams.microsoft.com/l/message/19:pHHeJGFqrgBu2fu0OhiVEy0y3MwXUEf-OwbxNf9BacM1@thread.tacv2/1758231367199?tenantId=a8eec281-aaa3-4dae-ac9b-9a398b9215e7&groupId=1d28e42e-f247-4360-854b-edf2109c5c15&parentMessageId=1758231367199&teamName=CS%205010%20(Fall%202025%2C%20Oakland)&channelName=General&createdTime=1758231367199

dependabot[bot] committed:
Chore(deps): Bump org.jetbrains:annotations in /versions (#13998)

Bumps [org.jetbrains:annotations](https://github.com/JetBrains/java-annotations) from 26.0.2 to 26.0.2-1.
- [Release notes](https://github.com/JetBrains/java-annotations/releases)
- [Changelog](https://github.com/JetBrains/java-annotations/blob/master/CHANGELOG.md)
- [Commits](https://github.com/JetBrains/java-annotations/compare/26.0.2...26.0.2-1)

---
updated-dependencies:
- dependency-name: org.jetbrains:annotations
  dependency-version: 26.0.2-1
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Chore(deps): Bump org.openrewrite.recipe:rewrite-recipe-bom (#13992)

Bumps [org.openrewrite.recipe:rewrite-recipe-bom](https://github.com/openrewrite/rewrite-recipe-bom) from 3.14.0 to 3.15.0.
- [Release notes](https://github.com/openrewrite/rewrite-recipe-bom/releases)
- [Commits](https://github.com/openrewrite/rewrite-recipe-bom/compare/v3.14.0...v3.15.0)

---
updated-dependencies:
- dependency-name: org.openrewrite.recipe:rewrite-recipe-bom
  dependency-version: 3.15.0
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Oliver Kopp <kopp.dev@gmail.com>

dependabot[bot] committed:
Chore(deps): Bump jablib/src/main/resources/csl-styles (#13999)

Bumps [jablib/src/main/resources/csl-styles](https://github.com/citation-style-language/styles) from `02faf1a` to `70bc1af`.
- [Release notes](https://github.com/citation-style-language/styles/releases)
- [Commits](https://github.com/citation-style-language/styles/compare/02faf1a0cc0885892363f0a7e4b7188a5ec0c25b...70bc1afb959a4defec5c2078593de4b1569820d5)

---
updated-dependencies:
- dependency-name: jablib/src/main/resources/csl-styles
  dependency-version: 70bc1afb959a4defec5c2078593de4b1569820d5
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Chore(deps): Bump commons-io:commons-io in /versions (#14001)

Bumps [commons-io:commons-io](https://github.com/apache/commons-io) from 2.19.0 to 2.20.0.
- [Changelog](https://github.com/apache/commons-io/blob/master/RELEASE-NOTES.txt)
- [Commits](https://github.com/apache/commons-io/compare/rel/commons-io-2.19.0...rel/commons-io-2.20.0)

---
updated-dependencies:
- dependency-name: commons-io:commons-io
  dependency-version: 2.20.0
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Chore(deps): Bump org.apache.commons:commons-lang3 in /versions (#14000)

Bumps org.apache.commons:commons-lang3 from 3.18.0 to 3.19.0.

---
updated-dependencies:
- dependency-name: org.apache.commons:commons-lang3
  dependency-version: 3.19.0
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Oliver Kopp <kopp.dev@gmail.com>

dependabot[bot] committed:
Chore(deps): Bump org.apache.logging.log4j:log4j-to-slf4j in /versions (#14002)

Bumps org.apache.logging.log4j:log4j-to-slf4j from 2.25.1 to 2.25.2.

---
updated-dependencies:
- dependency-name: org.apache.logging.log4j:log4j-to-slf4j
  dependency-version: 2.25.2
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Chore(deps): Bump io.zonky.test:embedded-postgres in /versions (#14004)

Bumps [io.zonky.test:embedded-postgres](https://github.com/zonkyio/embedded-postgres) from 2.1.0 to 2.1.1.
- [Release notes](https://github.com/zonkyio/embedded-postgres/releases)
- [Commits](https://github.com/zonkyio/embedded-postgres/compare/v2.1.0...v2.1.1)

---
updated-dependencies:
- dependency-name: io.zonky.test:embedded-postgres
  dependency-version: 2.1.1
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Chore(deps): Bump org.hisp.dhis:json-tree from 1.8 to 1.8.1 in /versions (#14005)

Bumps [org.hisp.dhis:json-tree](https://github.com/dhis2/json-tree) from 1.8 to 1.8.1.
- [Release notes](https://github.com/dhis2/json-tree/releases)
- [Changelog](https://github.com/dhis2/json-tree/blob/main/CHANGELOG.md)
- [Commits](https://github.com/dhis2/json-tree/commits)

---
updated-dependencies:
- dependency-name: org.hisp.dhis:json-tree
  dependency-version: 1.8.1
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Chore(deps): Bump net.java.dev.jna:jna-platform in /versions (#14006)

Bumps [net.java.dev.jna:jna-platform](https://github.com/java-native-access/jna) from 5.17.0 to 5.18.0.
- [Changelog](https://github.com/java-native-access/jna/blob/master/CHANGES.md)
- [Commits](https://github.com/java-native-access/jna/compare/5.17.0...5.18.0)

---
updated-dependencies:
- dependency-name: net.java.dev.jna:jna-platform
  dependency-version: 5.18.0
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

Nakul-Shivaraj committed:
Test: Refactor assigned test classes to use ParmeterizedTest

Nakul-Shivaraj committed:
Test: Refactor assigned test classes to use ParmeterizedTest

dependabot[bot] committed:
Chore(deps): Bump lucene from 10.2.2 to 10.3.0 in /versions (#13995)

* Chore(deps): Bump lucene from 10.2.2 to 10.3.0 in /versions

Bumps `lucene` from 10.2.2 to 10.3.0.

Updates `org.apache.lucene:lucene-analysis-common` from 10.2.2 to 10.3.0

Updates `org.apache.lucene:lucene-core` from 10.2.2 to 10.3.0

Updates `org.apache.lucene:lucene-highlighter` from 10.2.2 to 10.3.0

Updates `org.apache.lucene:lucene-queries` from 10.2.2 to 10.3.0

Updates `org.apache.lucene:lucene-queryparser` from 10.2.2 to 10.3.0

---
updated-dependencies:
- dependency-name: org.apache.lucene:lucene-analysis-common
  dependency-version: 10.3.0
  dependency-type: direct:production
  update-type: version-update:semver-minor
- dependency-name: org.apache.lucene:lucene-core
  dependency-version: 10.3.0
  dependency-type: direct:production
  update-type: version-update:semver-minor
- dependency-name: org.apache.lucene:lucene-highlighter
  dependency-version: 10.3.0
  dependency-type: direct:production
  update-type: version-update:semver-minor
- dependency-name: org.apache.lucene:lucene-queries
  dependency-version: 10.3.0
  dependency-type: direct:production
  update-type: version-update:semver-minor
- dependency-name: org.apache.lucene:lucene-queryparser
  dependency-version: 10.3.0
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

* change version to 103

* change version to 103

---------

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Siedlerchr <siedlerkiller@gmail.com>

Siedlerchr committed:
New Crowdin updates (#13985)

* New translations jabref_en.properties (French)

* New translations jabref_en.properties (Italian)

espertusnu commented on jablib/src/test/java/org/jabref/logic/layout/format/AuthorAndsReplacerTest.java:
```diff
@@ -10,15 +10,27 @@
     /**
-     * Test method for {@link org.jabref.logic.layout.format.AuthorAndsReplacer#format(java.lang.String)}.
+     * Test method for
+     * {@link org.jabref.logic.layout.format.AuthorAndsReplacer#format(java.lang.String)}.
```
Have you run javadoc and verified this generates a link and no warnings?

Nakul-Shivaraj replied:
No change made to the initial code, just added that to next line which I reverted back.

espertusnu commented on jablib/src/test/java/org/jabref/logic/layout/format/AuthorAndsReplacerTest.java:
```diff
@@ -10,15 +10,27 @@
-            "'John von Neumann and John Smith and Peter Black Brown', 'John von Neumann; John Smith & Peter Black Brown'"
+
+            // 'and' replaced with ';' and '&' for three names
+            "'von Neumann, John and Smith, John and Black Brown, Peter', "
```
I see you've added string concatenation (the `+` operator). Can you see if they use that or just have long lines in existing tests with long strings?

Nakul-Shivaraj replied:
I made this change to get rid of GoogleCheck issues. I have reverted back the code to long lines as they follow.

espertusnu commented on jablib/src/test/java/org/jabref/logic/layout/format/AuthorLastFirstAbbreviatorTest.java:
```diff
@@ -1,62 +1,43 @@
-    void oneAuthorCommonName() {
-        assertEquals("Lastname, N. M.", abbreviate("Lastname, Name Middlename"));
-    }
+    @ParameterizedTest(name = "[{index}] input=\"{0}\", expected=\"{1}\"")
```
Could you please explain to me what this does and why you did it?

Nakul-Shivaraj replied:
One of my test case was failing and I couldn’t figure out why. This annotation made the test output include the input and expected values, which helped me debug more easily. I forgot to remove it after fixing the issue. 
Thanks for pointing it out!!!
I'll update the code

espertusnu commented on jablib/src/test/java/org/jabref/logic/layout/format/RemoveBracketsTest.java:
```diff
@@ -2,41 +2,37 @@
+    @ParameterizedTest
+    @CsvSource({
+            // Brace pair correctly removed
+            "'{some text}', 'some text'",
```
Why are you using single quotes here?

Please remove them wherever they are not needed.

Nakul-Shivaraj replied:
Code updated.

espertusnu commented on jablib/src/test/java/org/jabref/logic/layout/format/RemoveLatexCommandsFormatterTest.java:
```diff
@@ -1,61 +1,48 @@
 class RemoveLatexCommandsFormatterTest {
 
-    private RemoveLatexCommandsFormatter formatter;
+    private final RemoveLatexCommandsFormatter formatter = new RemoveLatexCommandsFormatter();
```
Could you tell me why you removed the setup method?

Nakul-Shivaraj replied:
There was no need for BeforeEach because the object behaves the same across all tests so declared it with final too. And there were no use of BeforeEach in any of the test files I saw.

espertusnu replied:
The change is okay only if RemoveLatexCommandsFormatter is immutable. If a test *could* change it (even if it doesn't), keep BeforeEach.

Nakul-Shivaraj committed:
Address review comments

Nakul-Shivaraj committed:
Address review comments

koppor committed:
Add more JBang tests (#14010)

Siedlerchr committed:
New translations jabref_en.properties (Portuguese, Brazilian) (#14013)

josephyim committed:
Improved BibTeX brace check with more information in error message (#14012)

* Add more context to the error message when unmatched curly brackets are found.

* Update changelog

* use String::formatted and StringJoiner instead

* Add unit test

* Update CHANGELOG.md

Co-authored-by: Subhramit Basu <subhramit.bb@live.in>

---------

Co-authored-by: Subhramit Basu <subhramit.bb@live.in>

josephyim committed:
Fixed test org.jabref.cli.ArgumentProcessorTest#checkConsistency (#14015)

InAnYan committed:
feat(adrs): add more template engines to the ADR-0039 (#14014)

* feat(adrs): add more template engines to the ADR-0039

* Revert "feat(adrs): add more template engines to the ADR-0039"

This reverts commit 897d99eabc7be88f2f9c8d7b951fab34b96466d8.

* feat(adrs): add more template engines to the ADR-0039 (properly)

* fix(submodules): fix submodules

dependabot[bot] committed:
Chore(deps): Bump jbangdev/jbang-action from 0.132.0 to 0.132.1 (#14022)

Bumps [jbangdev/jbang-action](https://github.com/jbangdev/jbang-action) from 0.132.0 to 0.132.1.
- [Release notes](https://github.com/jbangdev/jbang-action/releases)
- [Commits](https://github.com/jbangdev/jbang-action/compare/v0.132.0...v0.132.1)

---
updated-dependencies:
- dependency-name: jbangdev/jbang-action
  dependency-version: 0.132.1
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Chore(deps): Bump gradle/actions from 4 to 5 (#14021)

Bumps [gradle/actions](https://github.com/gradle/actions) from 4 to 5.
- [Release notes](https://github.com/gradle/actions/releases)
- [Commits](https://github.com/gradle/actions/compare/v4...v5)

---
updated-dependencies:
- dependency-name: gradle/actions
  dependency-version: '5'
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Chore(deps): Bump jablib/src/main/resources/csl-styles (#14024)

Bumps [jablib/src/main/resources/csl-styles](https://github.com/citation-style-language/styles) from `70bc1af` to `9542192`.
- [Release notes](https://github.com/citation-style-language/styles/releases)
- [Commits](https://github.com/citation-style-language/styles/compare/70bc1afb959a4defec5c2078593de4b1569820d5...9542192e85a85a4c0be5467debede6ba77170c81)

---
updated-dependencies:
- dependency-name: jablib/src/main/resources/csl-styles
  dependency-version: 9542192e85a85a4c0be5467debede6ba77170c81
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Chore(deps): Bump com.github.javaparser:javaparser-symbol-solver-core (#14029)

Bumps [com.github.javaparser:javaparser-symbol-solver-core](https://github.com/javaparser/javaparser) from 3.27.0 to 3.27.1.
- [Release notes](https://github.com/javaparser/javaparser/releases)
- [Changelog](https://github.com/javaparser/javaparser/blob/master/changelog.md)
- [Commits](https://github.com/javaparser/javaparser/compare/javaparser-parent-3.27.0...javaparser-parent-3.27.1)

---
updated-dependencies:
- dependency-name: com.github.javaparser:javaparser-symbol-solver-core
  dependency-version: 3.27.1
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Chore(deps): Bump jablib/src/main/resources/csl-locales (#14025)

Bumps [jablib/src/main/resources/csl-locales](https://github.com/citation-style-language/locales) from `c72e5c7` to `8c149db`.
- [Release notes](https://github.com/citation-style-language/locales/releases)
- [Commits](https://github.com/citation-style-language/locales/compare/c72e5c75c7c432c0268543dc76e5d4ac09b6d9d5...8c149db30089aa45956d3373dbb0b269f7bd104f)

---
updated-dependencies:
- dependency-name: jablib/src/main/resources/csl-locales
  dependency-version: 8c149db30089aa45956d3373dbb0b269f7bd104f
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Chore(deps): Bump org.mockito:mockito-core in /versions (#14028)

Bumps [org.mockito:mockito-core](https://github.com/mockito/mockito) from 5.19.0 to 5.20.0.
- [Release notes](https://github.com/mockito/mockito/releases)
- [Commits](https://github.com/mockito/mockito/compare/v5.19.0...v5.20.0)

---
updated-dependencies:
- dependency-name: org.mockito:mockito-core
  dependency-version: 5.20.0
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Chore(deps): Bump com.github.andygoossens:gradle-modernizer-plugin (#14027)

Bumps com.github.andygoossens:gradle-modernizer-plugin from 1.11.0 to 1.12.0.

---
updated-dependencies:
- dependency-name: com.github.andygoossens:gradle-modernizer-plugin
  dependency-version: 1.12.0
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Chore(deps): Bump org.junit.jupiter:junit-jupiter in /versions (#14030)

Bumps [org.junit.jupiter:junit-jupiter](https://github.com/junit-team/junit-framework) from 5.13.4 to 6.0.0.
- [Release notes](https://github.com/junit-team/junit-framework/releases)
- [Commits](https://github.com/junit-team/junit-framework/compare/r5.13.4...r6.0.0)

---
updated-dependencies:
- dependency-name: org.junit.jupiter:junit-jupiter
  dependency-version: 6.0.0
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Chore(deps): Bump org.cyclonedx.bom from 3.0.0-alpha-1 to 3.0.0 (#14026)

Bumps org.cyclonedx.bom from 3.0.0-alpha-1 to 3.0.0.

---
updated-dependencies:
- dependency-name: org.cyclonedx.bom
  dependency-version: 3.0.0
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Chore(deps): Bump peter-evans/create-or-update-comment from 4 to 5 (#14023)

Bumps [peter-evans/create-or-update-comment](https://github.com/peter-evans/create-or-update-comment) from 4 to 5.
- [Release notes](https://github.com/peter-evans/create-or-update-comment/releases)
- [Commits](https://github.com/peter-evans/create-or-update-comment/compare/v4...v5)

---
updated-dependencies:
- dependency-name: peter-evans/create-or-update-comment
  dependency-version: '5'
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Carl Christian Snethlage <50491877+calixtus@users.noreply.github.com>

MartinBuzogan committed:
Implemented OpenAlex  (#14020)

* implementation of OpenAlexFetcher , Added UnitTest

* updated changelog

* updated tests for OpenAlex

* Updated WebFetcher to correct SourceCode test & converted parser tests to multi-line strings & applied DoiCleanup instead of removing it manualy

* OpenAlexFetcherTest Reformated

* Changed annotation NotNull to NonNull

koppor committed:
Update Gradle Wrapper from 9.3.0-jabref-1 to 9.1.0 (#14033)

Signed-off-by: gradle-update-robot <gradle-update-robot@regolo.cc>
Co-authored-by: gradle-update-robot <gradle-update-robot@regolo.cc>

koppor committed:
Enable update in versions of jablib (#14036)

Siedlerchr committed:
Fix eadr missing (#14043)

* add eadr

* preserver module

* export all packages

dependabot[bot] committed:
Chore(deps): Bump lucene from 10.3.0 to 10.3.1 in /versions (#14042)

Bumps `lucene` from 10.3.0 to 10.3.1.

Updates `org.apache.lucene:lucene-analysis-common` from 10.3.0 to 10.3.1

Updates `org.apache.lucene:lucene-core` from 10.3.0 to 10.3.1

Updates `org.apache.lucene:lucene-highlighter` from 10.3.0 to 10.3.1

Updates `org.apache.lucene:lucene-queries` from 10.3.0 to 10.3.1

Updates `org.apache.lucene:lucene-queryparser` from 10.3.0 to 10.3.1

---
updated-dependencies:
- dependency-name: org.apache.lucene:lucene-analysis-common
  dependency-version: 10.3.1
  dependency-type: direct:production
  update-type: version-update:semver-patch
- dependency-name: org.apache.lucene:lucene-core
  dependency-version: 10.3.1
  dependency-type: direct:production
  update-type: version-update:semver-patch
- dependency-name: org.apache.lucene:lucene-highlighter
  dependency-version: 10.3.1
  dependency-type: direct:production
  update-type: version-update:semver-patch
- dependency-name: org.apache.lucene:lucene-queries
  dependency-version: 10.3.1
  dependency-type: direct:production
  update-type: version-update:semver-patch
- dependency-name: org.apache.lucene:lucene-queryparser
  dependency-version: 10.3.1
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Christoph <siedlerkiller@gmail.com>

dependabot[bot] committed:
Chore(deps): Bump com.dlsc.pdfviewfx:pdfviewfx in /versions (#14040)

Bumps [com.dlsc.pdfviewfx:pdfviewfx](https://github.com/dlsc-software-consulting-gmbh/PDFViewFX) from 3.1.1 to 3.3.2.
- [Release notes](https://github.com/dlsc-software-consulting-gmbh/PDFViewFX/releases)
- [Changelog](https://github.com/dlsc-software-consulting-gmbh/PDFViewFX/blob/master/jreleaser.yml)
- [Commits](https://github.com/dlsc-software-consulting-gmbh/PDFViewFX/compare/v3.1.1...v3.3.2)

---
updated-dependencies:
- dependency-name: com.dlsc.pdfviewfx:pdfviewfx
  dependency-version: 3.3.2
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Christoph <siedlerkiller@gmail.com>

dependabot[bot] committed:
Chore(deps): Bump jablib/src/main/resources/csl-styles (#14037)

Bumps [jablib/src/main/resources/csl-styles](https://github.com/citation-style-language/styles) from `9542192` to `b61592e`.
- [Release notes](https://github.com/citation-style-language/styles/releases)
- [Commits](https://github.com/citation-style-language/styles/compare/9542192e85a85a4c0be5467debede6ba77170c81...b61592ea58b94d790fa36708048153989840552c)

---
updated-dependencies:
- dependency-name: jablib/src/main/resources/csl-styles
  dependency-version: b61592ea58b94d790fa36708048153989840552c
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Christoph <siedlerkiller@gmail.com>

dependabot[bot] committed:
Chore(deps): Bump org.mockito:mockito-core in /jablib (#14038)

Bumps [org.mockito:mockito-core](https://github.com/mockito/mockito) from 5.18.0 to 5.20.0.
- [Release notes](https://github.com/mockito/mockito/releases)
- [Commits](https://github.com/mockito/mockito/compare/v5.18.0...v5.20.0)

---
updated-dependencies:
- dependency-name: org.mockito:mockito-core
  dependency-version: 5.20.0
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Christoph <siedlerkiller@gmail.com>

dependabot[bot] committed:
Chore(deps): Bump com.vanniktech.maven.publish from 0.32.0 to 0.34.0 in /jablib (#14039)

* Chore(deps): Bump com.vanniktech.maven.publish in /jablib

Bumps [com.vanniktech.maven.publish](https://github.com/vanniktech/gradle-maven-publish-plugin) from 0.32.0 to 0.34.0.
- [Release notes](https://github.com/vanniktech/gradle-maven-publish-plugin/releases)
- [Changelog](https://github.com/vanniktech/gradle-maven-publish-plugin/blob/main/CHANGELOG.md)
- [Commits](https://github.com/vanniktech/gradle-maven-publish-plugin/compare/0.32.0...0.34.0)

---
updated-dependencies:
- dependency-name: com.vanniktech.maven.publish
  dependency-version: 0.34.0
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

* remove host

---------

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Christoph <siedlerkiller@gmail.com>

ThiloteE committed:
Rephrase JabRef-Machine's "Determine issue number" (#14044)

* Rephrase JabRef-Machine's "Determine issue number"

Removes unnecessary text, while retaining semantic similarity.

* Update .github/ghprcomment.yml

Co-authored-by: Subhramit Basu <subhramit.bb@live.in>

---------

Co-authored-by: Subhramit Basu <subhramit.bb@live.in>

espertusnu commented on jablib/src/test/java/org/jabref/logic/layout/format/AuthorFirstLastOxfordCommasTest.java:
```diff
@@ -10,25 +10,22 @@
-                    + "'John von Neumann, John Smith, and Peter Black Brown'",
-            "'John von Neumann and John Smith and Black Brown, Peter', "
-                    + "'John von Neumann, John Smith, and Peter Black Brown'"
+            "'von Neumann, John and Smith, John and Black Brown, Peter', 'John von Neumann, John Smith, and Peter Black Brown'",
```
I like these changes

espertusnu commented on jablib/src/test/java/org/jabref/logic/layout/format/AuthorLastFirstAbbrOxfordCommasTest.java:
```diff
@@ -10,26 +10,22 @@
 
             // Single Names
-            "'Van Something Someone', 'Someone, V. S.'",
+            "Van Something Someone, 'Someone, V. S.'",
```
I think it's okay to leave single quotes around one CSV item if the other one needs quoting, but what you've done is okay too.

espertusnu reviewed (commented):
I apologize for letting this slide. Feel free to check in with me 48 hours after requesting a re-review.

espertusnu commented:
When you're able to run javadoc, please report whether the @link generates a warning and/or produces a link.

renovate-bot committed:
Update dependency org.eclipse.jgit:org.eclipse.jgit.pgm to v7.4.0.202509020913-r (#14046)

Akanksha928 committed:
Automatic field editor: Add Clear content tab + viewmodel (#13824)

* Automatic field editor: Add Clear content tab + viewmodel

* Removing comments

* PR comment changes

* Bot suggested changes

* Bot suggested changes

* reformat

* fix fxml

* Added tests, reformats and visual modifications. Fixed number of affected files

* Fixed tab number

* CHANGELOG.md

---------

Co-authored-by: Siedlerchr <siedlerkiller@gmail.com>
Co-authored-by: Carl Christian Snethlage <calixtus@users.noreply.github.com>

Siedlerchr committed:
Switch to new macos intel runner and update arm runner as well (#14048)

* Switch to new macos intel runner and update arm runner as well

Refs https://github.blog/changelog/2025-09-19-github-actions-macos-13-runner-image-is-closing-down/

* macos15

* fix target names

* fix target names

espertusnu commented:
I didn't see an answer to my question about `@link`.

espertusnu commented:
I don't see any changes since my last review. Did you forget to push before making your re-review request?

Nakul-Shivaraj committed:
Merge branch 'issue676' of https://github.com/Nakul-Shivaraj/jabref into issue676

Nakul-Shivaraj committed:
Address review comment on BeforeEach

Nakul-Shivaraj commented:
Was able to run ./gradlew javadoc and the build was successful. No javadoc issues.

espertusnu commented:
Did the link work in the generated javadoc? You need to open the HTML to see.

espertusnu commented on jablib/src/test/java/org/jabref/logic/bibtex/FieldWriterTest.java:
```diff
@@ -126,12 +126,19 @@ void reportUnbalancedBracing() {
         assertThrows(InvalidFieldValueException.class, () -> writer.write(new UnknownField("anyfield"), unbalanced));
     }
 
+    @Test
```
Can you combine this with the preceding two methods with a ParameterizedTest?

Nakul-Shivaraj replied:
Not the file that I worked on

espertusnu commented on jablib/src/test/java/org/jabref/logic/citationkeypattern/BracketedPatternTest.java:
```diff
@@ -385,6 +385,15 @@ void expandBracketsWithFallback(String expandResult, String pattern) {
         assertEquals(expandResult, bracketedPattern.expand(bibEntry));
     }
 
+    @Test
```
Could you help me understand why you added this test? I don't see it in the original file.

Nakul-Shivaraj replied:
Not the file that I worked on

espertusnu commented on jablib/src/test/java/org/jabref/logic/externalfiles/LinkedFileHandlerTest.java:
```diff
@@ -67,8 +67,8 @@ void renameFile(String expectedFileName, String originalFileName, String newFile
             "file.pdf, https://example.com/file.pdf, ''",
-            "-.pdf, https://example.com/, pdf",
-            "-.pdf, path/to/file.pdf, pdf",
+            "file.pdf, https://example.com/, pdf",
```
Why did you make this change (or was that someone else's commit)?

Nakul-Shivaraj replied:
Not the file that I worked on

espertusnu commented on jablib/src/test/java/org/jabref/logic/layout/format/RemoveBracketsTest.java:
```diff
@@ -3,40 +3,41 @@
-        assertEquals("some text", formatter.format("some text"));
+            // Without brackets unmodified
+            "some text, some text"
+    })
```
Very good

espertusnu commented on jablib/src/test/java/org/jabref/logic/layout/format/RemoveLatexCommandsFormatterTest.java:
```diff
@@ -14,48 +15,39 @@ void setUp() {
-        assertEquals("http://pi.informatik.uni-siegen.de/stt/36_2/./03_Technische_Beitraege/ZEUS2016/beitrag_2.pdf", formatter.format("http://pi.informatik.uni-siegen.de/stt/36\\_2/./03\\_Technische\\_Beitraege/ZEUS2016/beitrag\\_2.pdf"));
+            // Realistic LaTeX URL with escaped underscores
+            "'http://pi.informatik.uni-siegen.de/stt/36\\_2/./03\\_Technische\\_Beitraege/ZEUS2016/beitrag\\_2.pdf', 'http://pi.informatik.uni-siegen.de/stt/36_2/./03_Technische_Beitraege/ZEUS2016/beitrag_2.pdf'"
+    })
```
Very good

espertusnu commented on jablib/src/test/java/org/jabref/logic/util/io/FileUtilTest.java:
```diff
@@ -82,7 +82,7 @@ void getLinkedFileNameDefaultFullTitle() {
 
         assertEquals("1234 - mytitle",
-                FileUtil.createFileNameFromPattern(null, entry, fileNamePattern));
+                FileUtil.createFileNameFromPattern(null, entry, fileNamePattern).get());
```
Were the changes to this file yours or someone else's?

Nakul-Shivaraj replied:
Not the file that I worked on

espertusnu reviewed (approved):
Approved, but do not merge.

espertusnu commented:
@Nakul-Shivaraj I generated the javadoc but was not able to find the page generated for your class. Could you please give me its path?

espertusnu commented:
If you look at the PR, it shows 77 changed files, which makes it hard for me to review. I believe that this is partly because you rebased and pushed your issue676 branch but not your main branch.

espertusnu commented:
I recommend that you do the following, especially because I approved this:
```
git checkout main 
git pull upstream main # to incorporate the latest changes from jabref's main branch
git checkout -b issue676-adds
git checkout issue676 [put here the paths to the files you changed]
git status # make sure you got the right files
git commit -am "Add parameterized tests"
git push -u origin issue676-adds
```
Then create a new draft PR
